### PR TITLE
BI-6944: Instagram capability

### PR DESCRIPTION
**Pull request by rajivraman**
Note that our Travis tests will fail until this is pushed to the core prod API.

### CODE REVIEW

File: test/Influencers/InfluencersTest.php [hunk 1]
```diff
@@ -292,6 +292,146 @@ public function testAddTwitterMissingParams()
         Traackr\Influencers::addTwitter([]);
     }
 
+    /**
```
Comment (rajivraman) [inline test/Influencers/InfluencersTest.php]: These tests are mainly copied from existing Twitter based tests. I'll point out differences below. If we ever deprecate/remove the old twitter methods, we can purge the duplication.

File: test/Influencers/InfluencersTest.php [hunk 2]
```diff
@@ -292,6 +292,146 @@ public function testAddTwitterMissingParams()
...
+
+        // Test based on Twitter ID type parameter
+        $twitter = Traackr\Influencers::lookupTwitter('7772342', 'TWITTER_ID');
+        $this->assertJsonStringEqualsJsonString(
+            json_encode($inf['influencer'][$this->infUid]),
+            json_encode($twitter['influencer']['7772342'])
+        );
+    }
+
+    /**
```
Comment (rajivraman) [inline test/Influencers/InfluencersTest.php]: new method

File: lib/TraackrApi/Influencers.php [hunk 3]
```diff
@@ -81,6 +116,40 @@ public static function lookupTwitter($username, $type = 'USERNAME')
...
+    public static function addSocial($p = array())
+    {
+        $inf = new Influencers();
+
+        $p = $inf->addCustomerKey($p);
+        $inf->checkRequiredParams($p, array('platform', 'customer_key'));
+
+        // Validate business requirements
+        $platform = $p['platform'];
+        if ('TWITTER' !== $platform && 'INSTAGRAM' !== $platform) {
```
Comment (sprak3000) [inline lib/TraackrApi/Influencers.php]: Could move the checks up to the front of the method. No need to do any other work above it if they do not pass.

File: lib/TraackrApi/Influencers.php [hunk 4]
```diff
@@ -81,6 +116,40 @@ public static function lookupTwitter($username, $type = 'USERNAME')
...
+            throw new \UnexpectedValueException('Platform parameter must be "TWITTER" or "INSTAGRAM".');
+        }
+        if (empty($p['username']) && empty($p['user_id'])) {
+            throw new MissingParameterException("Either username or user_id must be present");
+        } else if (!empty($p['username']) && !empty($p['user_id'])) {
```
Comment (sprak3000) [inline lib/TraackrApi/Influencers.php]: Can just be `if` given the previous conditional throws an exception.

File: test/Influencers/InfluencersTest.php [hunk 5]
```diff
@@ -292,6 +292,146 @@ public function testAddTwitterMissingParams()
         Traackr\Influencers::addTwitter([]);
     }
 
+    /**
+     * @group read-only
+     */
+    public function testLookupSocial()
```
Comment (sprak3000) [inline test/Influencers/InfluencersTest.php]: Would like to see a test to confirm the Instgram happy path works as well.

File: lib/TraackrApi/Influencers.php (hunk 3 above)
Comment (rajivraman) [reply lib/TraackrApi/Influencers.php]: i'd rather have them come in right after the checkRequiredParams() call. seems to be the norm we've settled on.

File: lib/TraackrApi/Influencers.php (hunk 4 above)
Comment (rajivraman) [reply lib/TraackrApi/Influencers.php]: that was a copy/paste but sure

File: test/Influencers/InfluencersTest.php (hunk 5 above)
Comment (rajivraman) [reply test/Influencers/InfluencersTest.php]: ok

**Comment by rajivraman** [comment]
@sprak3000 I took care of your feedback. Unit tests run fine locally.

**Pull request merged**